crownstone_core/util/cuckoofilter.py

```diff
@@ -0,0 +1,390 @@
+
+    # -------------------------------------------------------------
+
+    def add(self, key):
+        """
+        key: ByteArrayType
+        returns: bool
+        """
+        return self.addExtendedFingerprint(self.getExtendedFingerprint(key))
+
+    def remove(self, key):
+        """
+        key: ByteArrayType
+        returns: bool
+        """
+        return self.removeExtendedFingerprint(self.getExtendedFingerprint(key))
+
+    def contains(self, key):
+        """
+        key: ByteArrayType
+        returns: bool
+        """
+        return self.containsExtendedFingerprint(self.getExtendedFingerprint(key))
+
+
+    # -------------------------------------------------------------
+    # Init/deinit like stuff.
+    # -------------------------------------------------------------
+
+    def clear(self):
+        """
+        returns: None
+        """
+        self.victim = CuckooFilter.ExtendedFingerprint(0,0,0)
+        self.bucket_array = [0x00] * CuckooFilter.getfingerprintcount(self.bucket_count, self.nests_per_bucket)
+
+    # -------------------------------------------------------------
+    # Size stuff.
+    # -------------------------------------------------------------
+
+    @staticmethod
+    def sizeof(typ):
+        size_dict = {
+            'uint8': 1,
+            'uint16': 2,
+            'uint32': 4,
+            'uint64': 8,
+            'CuckooFilter': 1 + 1 + 2,
+            'FingerprintType': 2,
+            'IndexType': 1
+        }
+        if typ in size_dict:
+            return size_dict[typ]
+        return -1
+
+    @staticmethod
+    def getfingerprintcount(bucket_count, nests_per_bucket):
+        """
+        bucket_count: IndexType
+        nests_per_bucket: IndexType
+        returns: int
+        """
+        return bucket_count * nests_per_bucket
+
+    @staticmethod
+    def getbuffersize(bucket_count, nests_per_bucket):
+        """
+        bucket_count: IndexType
+        nests_per_bucket: IndexType
+        returns: int
+        """
+        return CuckooFilter.getfingerprintcount(bucket_count, nests_per_bucket) * CuckooFilter.sizeof('FingerprintType')
+
+    @staticmethod
+    def getsize(bucket_count, nests_per_bucket):
+        """
+        bucket_count: IndexType
+        nests_per_bucket: IndexType
+        returns: int
+        """
+        return CuckooFilter.sizeof('CuckooFilter') + CuckooFilter.getbuffersize(bucket_count, nests_per_bucket)
+
+    def fingerprintcount(self):
+        """
+        returns: int
+        """
+        return CuckooFilter.getfingerprintcount(self.bucket_count, self.nests_per_bucket)
+
+    def buffersize(self):
+        """
+        returns: int
+        """
+        return CuckooFilter.getbuffersize(self.bucket_count, self.nests_per_bucket)
+
+    def size(self):
+        """
+        returns: int
+        """
+        return CuckooFilter.getsize(self.bucket_count, self.nests_per_bucket)
+
+
+if __name__ == "__main__":
+    f = CuckooFilter(50, 4)
```

Comment:
this should be moved to a test file

Reply:
Check. See d456070150b93bdd18d23e8187977b6b10de7c85